admin/class-openedx-woocommerce-plugin-admin.php

```diff
@@ -336,4 +336,121 @@ public function save_custom_product_fields( $post_id ) {
 		update_post_meta( $post_id, '_course_id', $course_id );
 		update_post_meta( $post_id, '_mode', $mode );
 	}
+
+	/**
+	 * This function is called when an order payment is completed.
+	 *
+	 * @param int $order_id Order id.
+	 *
+	 * @return void
+	 */
+	public function process_order_data( $order_id ) {
+
+		$order         = wc_get_order( $order_id );
+		$billing_email = $order->get_billing_email();
+		$status        = $order->get_status();
+		$enrollment_id = '';
+		$courses       = array();
+
+		if ( 'processing' === $status ) {
+
+			$courses = $this->select_course_items( $order, $billing_email );
+
+			if ( ! empty( $courses ) ) {
+				wc_create_order_note( $order_id, 'Order items that are courses, obtained. ' );
+				$enrollment_id = $this->items_enrollment_request( $courses, $order_id, $billing_email );
+			}
+		}
+	}
+
+	/**
+	 * Select the items that are courses in the order.
+	 *
+	 * @param object $order Order object.
+	 *
+	 * @return array $courses Array of courses.
+	 */
+	public function select_course_items( $order ) {
+
+		$items   = $order->get_items();
+		$courses = array();
+
+		foreach ( $items as $item_id => $item ) {
+			$product_id = $item->get_product_id();
+			$course_id  = get_post_meta( $product_id, '_course_id', true );
+
+			if ( '' !== $course_id ) {
+				$courses[] = array( $item, $item_id );
+			}
+		}
+
+		return $courses;
+	}
+
+	/**
+	 * This function processes the enrollment request for each course in the order.
+	 *
+	 * @param array  $courses Array of courses.
+	 * @param int    $order_id Order id.
+	 * @param string $billing_email Billing email.
+	 *
+	 * @return void
+	 */
+	public function items_enrollment_request( $courses, $order_id, $billing_email ) {
+
+		foreach ( $courses as $item_id => $item ) {
+
+			$course_id    = get_post_meta( $item[0]->get_product_id(), '_course_id', true );
+			$course_mode  = get_post_meta( $item[0]->get_product_id(), '_mode', true );
+			$request_type = 'enroll';
+			$action       = 'enrollment_process';
+
+			$enrollment_arr = array(
+				'enrollment_course_id'    => $course_id,
+				'enrollment_email'        => $billing_email,
+				'enrollment_mode'         => $course_mode,
+				'enrollment_request_type' => $request_type,
+				'enrollment_order_id'     => $order_id,
+			);
+
+			$enrollment_id = $this->openedx_enrollment->insert_new( $enrollment_arr, $action, $order_id );
+			update_post_meta( $order_id, 'enrollment_id' . $item[1], $enrollment_id->ID );
```

Comment:
Same here : Instead of getting the information like this `$item[1]` which does not mean much changed it to something like `$item->enrollement_id;` by not creating an array with the information store under positions that are not clear to get, please refactor and use an object to store the data so when using it, it gives more meaning to what is being fetch